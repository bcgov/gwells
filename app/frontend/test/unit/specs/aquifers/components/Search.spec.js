import { shallowMount, createLocalVue } from '@vue/test-utils'
import SearchComponent from '@/aquifers/components/Search.vue'
import { SEARCH_AQUIFERS, RESET_RESULTS } from '@/aquifers/store/actions.types'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

const aquiferFixture = {
  aquifer_id: '4',
  aquifer_name: 'Aggasiz',
  location_description: 'Aggasiz/Sea Bird Island',
  material_description: 'Sand and Gravel',
  litho_stratographic_unit: 'Fraser River Sediments',
  subtype_description: 'Unconfined sand and gravel - large river system',
  vulnerability_description: 'High',
  area: '63.3',
  productivity_description: 'High',
  demand: 'Low',
  mapping_year: '1993'
}

describe('Search Component', () => {
  const component = ({ getters, actions }) => shallowMount(SearchComponent, {
    store: new Vuex.Store({
      getters: Object.assign({
        aquiferList: () => [],
        emptyResults: () => false
      }, getters),
      actions
    }),
    localVue
  })

  it('Displays a message if no aquifers could be found', () => {
    const wrapper = component({
      getters: { emptyResults: () => true }
    })

    expect(wrapper.text()).toContain('No aquifers could be found')
  })

  it('Matches the snapshot', () => {
    const wrapper = component({
      getters: { aquiferList: () => [aquiferFixture] }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('triggers SEARCH_AQUIFERS when form is submitted', () => {
    const spy = jest.fn()
    const wrapper = component({
      actions: { [SEARCH_AQUIFERS]: spy }
    })

    wrapper.find('form').trigger('submit')
    expect(spy).toHaveBeenCalled()
  })

  it('triggers RESET_RESULTS when form is reset', () => {
    const spy = jest.fn()
    const wrapper = component({
      actions: { [RESET_RESULTS]: spy }
    })

    wrapper.find('form').trigger('reset')
    expect(spy).toHaveBeenCalled()
  })

  it('resets input values when form is reset', () => {
    const spy = jest.fn()
    const wrapper = component({
      actions: { [RESET_RESULTS]: spy }
    })

    wrapper.find('form').trigger('reset')
    expect(spy).toHaveBeenCalled()
  })
})
