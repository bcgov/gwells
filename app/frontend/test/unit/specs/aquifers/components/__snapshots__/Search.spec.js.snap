// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Matches the snapshot 1`] = `
<div
  class="card p-3 mb-4"
>
  <h5>
    Aquifer Search
  </h5>
   
  <form
    class=""
  >
    <div
      class="form-row"
    >
      <div
        class="col-md-4 col-12"
      >
        <fieldset
          aria-labelledby="__BVID__14__BV_label_"
          class="b-form-group form-group"
          id="__BVID__14"
          role="group"
        >
          <legend
            class="col-form-label pt-0"
            id="__BVID__14__BV_label_"
          >
            Aquifer name
          </legend>
          <div
            aria-labelledby="__BVID__14__BV_label_"
            class=""
            role="group"
          >
            <input
              class="form-control"
              id="__BVID__15"
              type="text"
            />
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
         
        <fieldset
          aria-labelledby="__BVID__16__BV_label_"
          class="b-form-group form-group"
          id="__BVID__16"
          role="group"
        >
          <legend
            class="col-form-label pt-0"
            id="__BVID__16__BV_label_"
          >
            Aquifer number
          </legend>
          <div
            aria-labelledby="__BVID__16__BV_label_"
            class=""
            role="group"
          >
            <input
              class="form-control"
              id="__BVID__17"
              type="text"
            />
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
      </div>
    </div>
     
    <div
      class="form-row"
    >
      <div
        class="col"
      >
        <fieldset
          class="b-form-group form-group"
          id="__BVID__18"
          role="group"
        >
          <!---->
          <div
            class=""
            role="group"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Search
            </button>
             
            <button
              class="btn btn-default"
              type="reset"
            >
              Reset
            </button>
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
      </div>
    </div>
  </form>
   
  <table
    aria-busy="false"
    aria-colcount="12"
    aria-rowcount="1"
    class="table b-table"
    id="__BVID__19"
  >
    <!---->
    <!---->
    <thead
      class=""
    >
      <tr>
        <th
          aria-colindex="1"
          class=""
        >
          Aquifer number
        </th>
        <th
          aria-colindex="2"
          class=""
        >
          Aquifer name
        </th>
        <th
          aria-colindex="3"
          class=""
        >
          Descriptive location
        </th>
        <th
          aria-colindex="4"
          class=""
        >
          Material
        </th>
        <th
          aria-colindex="5"
          class=""
        >
          Litho stratigraphic unit
        </th>
        <th
          aria-colindex="6"
          class=""
        >
          Subtype
        </th>
        <th
          aria-colindex="7"
          class=""
        >
          Vulnerability
        </th>
        <th
          aria-colindex="8"
          class=""
        >
          Size-kmÂ²
        </th>
        <th
          aria-colindex="9"
          class=""
        >
          Productivity
        </th>
        <th
          aria-colindex="10"
          class=""
        >
          Demand
        </th>
        <th
          aria-colindex="11"
          class=""
        >
          Year of mapping
        </th>
        <th
          aria-colindex="12"
          class=""
        >
          Publication status
        </th>
      </tr>
    </thead>
    <!---->
    <tbody
      class=""
    >
      <!---->
      <tr
        class=""
      >
        <td
          aria-colindex="1"
          class=""
        >
          4
        </td>
        <td
          aria-colindex="2"
          class=""
        >
          Aggasiz
        </td>
        <td
          aria-colindex="3"
          class=""
        >
          Aggasiz/Sea Bird Island
        </td>
        <td
          aria-colindex="4"
          class=""
        >
          Sand and Gravel
        </td>
        <td
          aria-colindex="5"
          class=""
        >
          Fraser River Sediments
        </td>
        <td
          aria-colindex="6"
          class=""
        >
          Unconfined sand and gravel - large river system
        </td>
        <td
          aria-colindex="7"
          class=""
        >
          High
        </td>
        <td
          aria-colindex="8"
          class=""
        >
          63.3
        </td>
        <td
          aria-colindex="9"
          class=""
        >
          High
        </td>
        <td
          aria-colindex="10"
          class=""
        />
        <td
          aria-colindex="11"
          class=""
        >
          1993
        </td>
        <td
          aria-colindex="12"
          class=""
        />
      </tr>
      <!---->
      <!---->
    </tbody>
  </table>
</div>
`;
